pandas/tests/io/json/test_pandas.py

```diff
@@ -2162,30 +2155,18 @@ def test_read_json_dtype_backend(
             }
         )
 
-        if using_infer_string:
-            string_array = ArrowStringArrayNumpySemantics(pa.array(["a", "b", "c"]))
-            string_array_na = ArrowStringArrayNumpySemantics(pa.array(["a", "b", None]))
-        elif string_storage == "python":
-            string_array = StringArray(np.array(["a", "b", "c"], dtype=np.object_))
-            string_array_na = StringArray(np.array(["a", "b", NA], dtype=np.object_))
-
-        elif dtype_backend == "pyarrow":
-            pa = pytest.importorskip("pyarrow")
-            from pandas.arrays import ArrowExtensionArray
-
-            string_array = ArrowExtensionArray(pa.array(["a", "b", "c"]))
-            string_array_na = ArrowExtensionArray(pa.array(["a", "b", None]))
-
-        else:
-            string_array = ArrowStringArray(pa.array(["a", "b", "c"]))
-            string_array_na = ArrowStringArray(pa.array(["a", "b", None]))
-
         out = df.to_json(orient=orient)
         with pd.option_context("mode.string_storage", string_storage):
             result = read_json(
                 StringIO(out), dtype_backend=dtype_backend, orient=orient
             )
 
+        if dtype_backend == "pyarrow":
+            pa = pytest.importorskip("pyarrow")
+            string_dtype = pd.ArrowDtype(pa.string())
+        else:
+            string_dtype = pd.StringDtype(string_storage)
```

Comment:
This is actually a different logic than the lines that are removed just above, i.e. it no longer uses `using_infer_string`, because when the user explicitly passes `dtype_backend="pyarrow"|"numpy_nullable"`, the user should actually always get the NA-variants of the string dtype (regardless of whether the future (NaN-based) string dtype is enabled or not).

Reply:
This is probably useful in quite a few places in testing right? Maybe even in the core codebase? I wonder if this shouldn't be a helper function instead, or maybe the StringDtype `__new__` should handle this and return a pd.ArrowDtype for the pyarrow backend.

Kind of an in between spot since `pd.ArrowDtype` is still a separate concept from `pd.StringDtype`, but I think in the future will be easier to refactor if we

Reply:
> This is probably useful in quite a few places in testing right? 

I haven't yet encountered many places in the code base itself that uses that logic, IIRC (although https://github.com/pandas-dev/pandas/pull/59487 is actually an example of where we maybe should do that, and the fact that we currently don't use ArrowDtype there is a bit of a bug/missing piece ..)

>  maybe the StringDtype __new__ should handle this and return a pd.ArrowDtype for the pyarrow backend.

Given those are still separate concepts, as you mention (with also different arrays with different behaviour in various places), I personally think it would make the current situation on the short term rather more confusing if some invocation of `StringDtype(..)` would return `ArrowDtype(string)` (it would also require a new keyword because `StringDtype("pyarrow")` already exists). And on the longer term we should see what the outcome will be for PDEP-13 on logical types.  
Especially for testing here, I think it is also good to be explicit about which dtype class is the expected one, for readability / understand-ability of the tests.

Reply:
Fair points. If we start seeing in the core library, I think a common function to generate this would make more sense